src/main/java/sh/okx/railswitch/DetectorRailActivateListener.java

```diff
@@ -0,0 +1,163 @@
+                }
+                if (!Objects.equals(
+                        ((PlayerReinforcement) railReinforcement).getGroup(),
+                        ((PlayerReinforcement) signReinforcement).getGroup())) {
+                    return;
+                }
+            }
+        }
+        
+        // Check that a player is triggering the switch
+        // NOTE: The event doesn't provide the information and so the next best thing is searching for a
+        //       player who is nearby and riding a minecart.
+        Player player = null; {
+            double searchDistance = Double.MAX_VALUE;
+            for (Entity entity : block.getWorld().getNearbyEntities(block.getLocation(), 3, 3, 3)) {
+                if (entity.getType() != EntityType.PLAYER
+                        || !(entity instanceof Player)) {
+                    continue;
+                }
+                Entity vehicle = entity.getVehicle();
+                if (vehicle == null
+                        || vehicle.getType() != EntityType.MINECART
+                        || !(vehicle instanceof Minecart)) {
+                    continue;
+                }
+                double distance = block.getLocation().distanceSquared(entity.getLocation());
+                if (distance < searchDistance) {
+                    searchDistance = distance;
+                    player = (Player) entity;
+                }
+            }
+        }
+        if (player == null) {
+            return;
+        }
+        
+        // Determine whether a player has a destination that matches one of the destinations
+        // listed on the switch signs, or match if there's a wildcard.
+        boolean matched = false;
+        String[] playerDestinations = plugin.getDatabase().
+                getPlayerDestination(player).
+                map(dest -> dest.split(" ")).
+                orElse(null);
+        if (playerDestinations != null && playerDestinations.length > 0) {
+            String[] switchDestinations = Arrays.copyOfRange(lines, 1, lines.length);
+            matcher:
+            for (String playerDestination : playerDestinations) {
+                if (playerDestination.equals(WILDCARD)) {
+                    matched = true;
+                    break;
+                }
+                for (String switchDestination : switchDestinations) {
+                    if (switchDestination.equals(WILDCARD)
+                            || playerDestination.equalsIgnoreCase(switchDestination)) {
+                        matched = true;
+                        break matcher;
+                    }
+                }
+            }
+        }
+        
+        switch (type) {
+            case NORMAL:
+                event.setNewCurrent(matched ? 15 : 0);
+                break;
+            case INVERTED:
+                event.setNewCurrent(matched ? 0 : 15);
+                break;
+            // Gotta have this or the Devoted style checker will screech
```

Comment:
Yes,  because type could be null

Reply:
Nope :P there's a null check further up the code

Reply:
Nope :P there's a null check further up the code